![GTPB](http://gtpb.igc.gulbenkian.pt/bicourses/images/GTPB2015logo.png "GTPB")
asdfasd
# ARANGS 2016

This is the git repository for the course *Automated and Reproducible Analysis of Next 
Generation Sequencing data* (ARANGS16). This course is organized by 
[Pedro Fernandes](https://github.com/pfern) and taught by [Darin London](https://github.com/dmlond),
[Hannes Hettling](http://github.com/hettling) and [Rutger Vos](https://github.com/rvosa) under the 
auspices of the [Gulbenkian Training Programme in Bioinformatics](http://gtpb.igc.gulbenkian.pt/bicourses/)
in Oeiras, Portugal.

* [Course Homepage](http://gtpb.igc.gulbenkian.pt/bicourses/ARANGS16)
* [Course Timetable](http://gtpb.igc.gulbenkian.pt/bicourses/ARANGS16/timetable.html)

This repository contains source code, data, configuration files, documentation and
reference materials. Unless indicated otherwise, all these materials can be reused
under the terms of the [MIT](https://github.com/rvosa/arangs2016/blob/master/LICENSE) license.

## Additional reading of interest

* [_The Case for Cloud Computing in Genome Informatics_](http://dx.doi.org/10.1186/gb-2010-11-5-207)
* [_A Quick Guide for Developing Effective Bioinformatics Programming Skills_](http://dx.doi.org/10.1371/journal.pcbi.1000589)
* [_A Quick Guide to Organizing Computational Biology Projects_](http://dx.doi.org/10.1371/journal.pcbi.1000424)
* [_Ten Simple Rules for Taking Advantage of git and GitHub_](http://dx.doi.org/10.1101/048744)
* [_Ten Simple Rules for Managing High-Throughput Nucleotide Sequencing Data_](http://dx.doi.org/10.1101/049338)
* Unix Tutorials: [General intro](http://www.ee.surrey.ac.uk/Teaching/Unix/), [Bash scripting](http://tldp.org/LDP/abs/html/)

<!--
## NGS reference materials

The following are somewhat old and a bit out of scope for the course, but might still be interesting:

* DNA Sequencing Technologies
http://www.nature.com/scitable/topicpage/DNA-Sequencing-Technologies-690

* "The Sanger FASTQ file format for sequences with quality scores, and the Solexa/Illumina FASTQ variants"
http://dx.doi.org/10.1093%2Fnar%2Fgkp1137

* "A Practical Comparison of De Novo Genome Assembly Software Tools for Next-Generation Sequencing Technologies"
http://dx.doi.org/10.1371/journal.pone.0017915

* "A beginner's guide to eukaryotic genome annotation"
http://dx.doi.org/10.1038/nrg3174

* NGS glossary spreadsheet
https://docs.google.com/spreadsheet/ccc?key=0Av8UW3JvZsgcdE9wZW1sYzlCQWFwNjBXLWMtQzZLN3c#gid=0

* NGS platforms
https://docs.google.com/document/pub?id=1rYbBPELjjezRVjkQfkulJI2jNxL5LsRuNXVv_CxCpd4

## Syntax Format Descriptions

* SAM/BAM http://samtools.sourceforge.net/SAM1.pdf
* VCF Format http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40
* FASTQ http://maq.sourceforge.net/fastq.shtml
* Sequence file formats http://bioinf.comav.upv.es/courses/sequence_analysis/sequence_file_formats.html

-->

## Tools we will use

### Executables

* samtools http://samtools.sourceforge.net/
* bwa http://bio-bwa.sourceforge.net/

### Docker, docker-machine, docker-compose

* docker https://www.docker.com/
* docker-machine https://docs.docker.com/machine/
* docker-compose https://docs.docker.com/compose/

### Vagrant, virtualbox

* vagrant http://vagrantup.com
* virtualbox http://virtualbox.org
